debian/libcephsqlite.install

```diff
@@ -0,0 +1 @@
+usr/lib/libcephsqlite.so*
```

Comment:
might want to install
```
usr/lib/libcephsqlite.so.*
```

instead. as libcephsqlites with different so versions can be colocated in future.

Reply:
I think I've tried that before but the symlink (`libcephsqlite.so`) does not get installed. Tests would fail on ubuntu because they could not find the library.

Reply:
tests should always link against a specific shared library, like libcephsqlite.so.1 instead of libcephsqlite.so.

Reply:
I don't think that practice is compatible with sqlite dynamic extension loading: https://github.com/ceph/ceph/pull/39191/files#diff-7c88009e7866da6cdb077a81ff2922ae91ef62af9c3a3decfdb0e3a82dc5c848R23

See also: https://sqlite.org/loadext.html

I actually had to be explicit about the `.so` extension in the above tests as the sqlite extension search would fail if I just did `.load libcephsqlite` (on Ubuntu, IIRC). Reviewing that doc again would indicate it should not be necessary:

>If you want to make your code portable, you can omit the suffix from the shared library filename...

For the tests, I plan to just keep the suffix but it looks like a bug in the Ubuntu packaging of sqlite.

Reply:
hmm, I will read the reference doc and get back to you. seems to load sqlite vfs extension is not like how we link against libsqlite3.so..

Reply:
TMK, this practice is common for libraries loaded dynamically at run-time via programming languages. It's done for Python and Lua. e.g.:

````
$ ls -l /usr/lib/python3.9/*/*so | head
-rwxr-xr-x 1 root root   65560 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/array.cpython-39-x86_64-linux-gnu.so
-rwxr-xr-x 1 root root   61880 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/_asyncio.cpython-39-x86_64-linux-gnu.so
-rwxr-xr-x 1 root root   43704 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/audioop.cpython-39-x86_64-linux-gnu.so
-rwxr-xr-x 1 root root   43872 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/binascii.cpython-39-x86_64-linux-gnu.so
-rwxr-xr-x 1 root root   14680 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/_bisect.cpython-39-x86_64-linux-gnu.so
-rwxr-xr-x 1 root root   48608 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/_blake2.cpython-39-x86_64-linux-gnu.so
-rwxr-xr-x 1 root root   27768 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/_bz2.cpython-39-x86_64-linux-gnu.so
-rwxr-xr-x 1 root root   68320 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/cmath.cpython-39-x86_64-linux-gnu.so
-rwxr-xr-x 1 root root  157640 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/_codecs_cn.cpython-39-x86_64-linux-gnu.so
-rwxr-xr-x 1 root root  161736 Feb 20 10:40 /usr/lib/python3.9/lib-dynload/_codecs_hk.cpython-39-x86_64-linux-gnu.so
 ls -l /usr/lib/lua/5.4/
total 188
-rwxr-xr-x 1 root root 114048 Mar  3 09:35 gv.so
-rwxr-xr-x 1 root root  22664 Jul 31  2020 lfs.so
-rwxr-xr-x 1 root root  51448 Jun 30  2020 lpeg.so
````

Reply:
so, in other words, they are like plugins, and hence should not be (so) versioned.

Reply:
make sense to me.